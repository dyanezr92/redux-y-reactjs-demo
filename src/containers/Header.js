import { connect } from 'react-redux'
import HeaderWithPicture from '../components/HeaderWithPicture'
import { getComments } from '../reducers/commentsReducer'
import { getComplaints } from '../reducers/complaintsReducer'

const mapStateToProps = (state, ownProps) => ({
  numComments: getComments(state).length,
  numComplaints: getComplaints(state).length,
  title: ownProps.title,
  src: ownProps.src
})

export default connect(
  mapStateToProps,
  null
)(HeaderWithPicture)
