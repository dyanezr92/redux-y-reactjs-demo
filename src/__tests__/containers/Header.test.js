import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components
import { Provider } from 'react-redux';

import Header from '../../containers/Header';
import finn from '../../static/finn.gif'

// Configurar el mock de la store
const mockStore = configureStore();
const initialState = {
  comments: [],
  complaints: []
};
const store = mockStore(initialState);

// TEST
describe('<Header />', () => {
  test('shows the header with the store data', () => {
    const wrapper = shallow(
        <Header store={ store } title="Awesome Finn" src={ finn } />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
