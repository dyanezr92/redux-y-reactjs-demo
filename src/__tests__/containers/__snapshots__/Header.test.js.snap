// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shows the header with the store data 1`] = `
<HeaderWithPicture
  dispatch={[Function]}
  numComments={0}
  numComplaints={0}
  src="finn.gif"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  title="Awesome Finn"
/>
`;
